File: CRC.cpp
#include <iostream>
#include <string>

using namespace std;

// Function to compute the CRC
string computeCRC(string input, string polynomial) {
    int n = input.length();
    int m = polynomial.length();

    string dividend = input + string(m - 1, '0');

    for (int i = 0; i <= n - 1; i++) {
        if (dividend[i] == '1') {
            for (int j = 0; j < m; j++) {
                dividend[i + j] = (dividend[i + j] == polynomial[j]) ? '0' : '1';
            }
        }
    }

    return dividend.substr(n, m - 1);
}

int main() {
    string input, polynomial;

    cout << "Enter the input binary string: ";
    cin >> input;
    cout << "Enter the polynomial binary string: ";
    cin >> polynomial;

    string crc = computeCRC(input, polynomial);
    string transmitted = input + crc;

    cout << "The computed CRC is: " << crc << endl;
    cout << "The transmitted message is: " << transmitted << endl;

    return 0;
}

File: CRC.py
def compute_crc(input_bits, polynomial_bits):
    n = len(input_bits)
    m = len(polynomial_bits)
    
    # Append m-1 zeros to the input bits
    dividend = input_bits + '0' * (m - 1)
    dividend = list(dividend)
    
    for i in range(n):
        if dividend[i] == '1':
            for j in range(m):
                dividend[i + j] = '0' if dividend[i + j] == polynomial_bits[j] else '1'
    
    return ''.join(dividend[-(m-1):])

if __name__ == "__main__":
    input_bits = input("Enter the input binary string: ")
    polynomial_bits = input("Enter the polynomial binary string: ")
    
    crc = compute_crc(input_bits, polynomial_bits)
    transmitted_message = input_bits + crc
    
    print("The computed CRC is:", crc)
    print("The transmitted message is:", transmitted_message)
